sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/BeamFnDataGrpcClient.java

```diff
@@ -82,6 +82,14 @@ public void unregisterReceiver(
     }
   }
 
+  @Override
+  public void poisonInstructionId(String instructionId) {
+    LOG.debug("Poisoning instruction {}", instructionId);
+    for (BeamFnDataGrpcMultiplexer client : multiplexerCache.values()) {
```

Comment:
one thing this doesn't yet handle is if the instruction that is being poisoned is for an endpoint that has not yet been observed.  This is only possible at beginning of the processing though.

I was working on refactoring creation of the bundle processor to determine the api endpoints before possible exception was thrown but it was getting large and would prefer to do it as a follow up because as-is this helps cases where the api endpoint was created before.

Reply:
+1 to getting this in first. 

An alternative is for all these multiplexers to share the same poisoned id cache.